configs/components/bolt-create-ruby-tarballs.rb

```diff
@@ -0,0 +1,36 @@
+component "bolt-create-ruby-tarballs" do |pkg, settings, platform|
+  # We only create the tarballs on Windows right now
+  if platform.is_windows?
+    pkg.add_source("file://resources/files/install-tarballs/extract_all.rb")
+    pkg.add_source("file://resources/files/install-tarballs/remove_all.rb")
+    pkg.directory "#{settings[:datadir]}/install-tarballs"
+    # Unlike other examples, where the path would be '../<file>' we use the current workding directory.  This is because
+    # even though we add a source as above, because it is not a gem or tar etc., the component has nothing to extract therefore
+    # we don't end up in a component directory
+    pkg.install_file "extract_all.rb", "#{settings[:datadir]}/install-tarballs/extract_all.rb"
+    pkg.install_file "remove_all.rb", "#{settings[:datadir]}/install-tarballs/remove_all.rb"
+
+    pkg.build do
+      build_commands = []
+
+      # Create the destination directory incase it doesn't exist already
+      build_commands << "mkdir -p #{settings[:datadir]}/install-tarballs"
+      build_commands << "pushd #{settings[:prefix]}"
+
+      delete_dirs = []
```

Comment:
Is this left over from something? Seems like there is nothing to delete.

Reply:
James can correct me if I'm wrong, but what this is doing is creating a tarball of the gems Bolt ships with and deleting the gems themselves once the tarball is created before packaging Bolt. Then on install, we'll untar the gems. This reduces the install size and time of the Bolt MSI package. This var specifically tracks the directories to tar then delete - we can probably more directly just use `dirs` since we don't build up multiple directories to tar the way PDK does

Reply:
Lucy is correct. It's alot more complicated for PDK as they track ~3 versions of ruby and all the gems that come with each version: https://github.com/puppetlabs/pdk-vanagon/blob/6f9daaef760cf68557ba7a23ace7be8f7fe800fc/configs/components/pdk-create-ruby-tarballs.rb#L29-L34. I was doing a 'remove what doesn't apply, but don't change functionality' approach, so changed very little code. 

There is only one directory in the list because I did try to add `share/doc` and `share/man`, which have about ~5000 files in them, but they are symlinks in the source and the custom minitar code we wrote in PDK specifically avoids symlink files. I didn't make a note of it in this PR, but I intended to come back and work that would. It would remove another second or two from the install time.

That being said, I'm fine with using `dirs` and then making another variable if more processing is needed.

Reply:
Modified to suit